### PR TITLE
Apply ci.yaml to all targets in the Makefile, not only top-level module

### PR DESCRIPTION
**Pull request by jmacd**
We would like the CI build and test stages to apply to all Go modules (including the collector and the otelarrowcol).

### CODE REVIEW

**Comment by jmacd** [comment]
See https://github.com/open-telemetry/otel-arrow/issues/39

**Review by moh-osman3 — APPROVED** [review]

**Comment by jmacd** [comment]
I've removed the collector, it was making this difficult and I want to take just the one step of adding the collector/... tests, now.

**Review by lquerel — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Comment by jmacd** [comment]
Not sure this was working.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,10 +21,10 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v3
       with:
-        go-version: 1.19
+        go-version: 1.20
```
Comment (moh-osman3) [inline .github/workflows/ci.yml]: Go versioning seems to be the issue here - found https://github.com/actions/setup-go/issues/326#issuecomment-1414855139 which ran into the same issue. The `1.20` resolves to 
``` 
Successfully set up Go version 1.2
go version go1.2.2 linux/amd64
 ```

This should be changed to `1.20.0` instead.  Unable to push directly to this branch but tested that it works by creating this dummy PR https://github.com/open-telemetry/otel-arrow/pull/64 and confirming all the collector component tests run now.

Comment (jmacd) [reply .github/workflows/ci.yml]: AHA I hadn't seen this note and of course yes, that explained it, and the fix is already upstream.

**Pull request merged**